docs/decisions/20240115-Error-handling-report-problem-agent.md

```diff
@@ -0,0 +1,166 @@
+involving collaboration among various parties for a solution.
+For DIDComm, the aspect of interoperability is especially critical in the context of problem reporting. 
+In DIDComm, an agent developed by one team is required to be adept at interpreting errors reported by an agent from a completely different team,
+presenting a unique challenge in this area.
+
+In Agent Currently  we have 3 flows Connection Issuance and Verification.
+Communication between 2 agents Agent A <-> Agent B happens in the background job in async
+Error occurred in the background job over didcomm Agent A is logged in Agent A logs
+But the Agent B is not aware of anything 
+
+## Decision Drivers <!-- optional -->
+
+What are our needs? Let’s try to sum up the required capabilities based on 
+[Report Problem 2.0](https://identity.foundation/didcomm-messaging/spec/#problem-reports), we need:
+
+* Find out all the possible scenarios on various levels for the 3 flows Connection Issuance and Verification
+
+When processing any step of any protocol, if something is wrong, the Agent tries to:
+Update the record to a documented error state
+Log the error in the service logs
+Send the problem report message when appropriate following return route
+https://github.com/decentralized-identity/didcomm-messaging/blob/main/extensions/return_route/main.md
+
+| Agent Flows           | Protocols                                                                                          |
+|-----------------------|----------------------------------------------------------------------------------------------------|
+| Connection            | https://github.com/hyperledger/aries-rfcs/tree/main/features/0160-connection-protocol              |
+| Issuance              | https://github.com/decentralized-identity/waci-didcomm/tree/main/issue_credential                  |
+| Verification(Present proof) | https://github.com/decentralized-identity/waci-didcomm/blob/main/present_proof/present-proof-v3.md |
+
+  Custom Behavior table
+  This table defines the expected behavior of the Agent in different scenarios not covered by the specifications.
+
+ | Agent            | Behaviour                      | Comments            |
+ |------------------|--------------------------------|---------------------|
+ | Scenario G1	     | Send a problem report          | e.p.msg.unsupported |
+ | Scenario G2      | Send a problem report          | e.p.msg.unsupported | 
+ | Scenario G3      | Send a problem report          | e.p.error           | 
+ | Scenario G4      | (Sync ?) Send a problem report | e.p.trust.crypto    | 
+ | Scenario G5      | (Sync ?) Send a problem report | e.p.did             | 
+ | Scenario G6      | (Sync ?) Send a problem report | e.p.did.malformed   | 
+ | Scenario G7      | Send a problem report          | e.p.msg.<PIURI>     | 
+
+### Scenarios Description
+
+#### General Scenarios:
```

Comment:
If you read this in the markdown viewer, it isn't that clear from the bold text alone that the first line is the input/trigger and the line after is the outcome/effect - I had to think very hard about it :-)